app/__init__.py

```diff
@@ -1,35 +1,34 @@
-from typing import List
-from flask import Flask, jsonify, request, Response
-import subprocess
-import glob
 import os
+import re
+import glob
 import yaml
-from flask_swagger_ui import get_swaggerui_blueprint
-import shutil
 import json
-import logging
-import re
 import uuid
+import shutil
+import logging
+import subprocess
+from response_messages import errors, response
 from logging.handlers import RotatingFileHandler
-import data_access_layer.RestDB
-from config import errors, response
+from flask import Flask, jsonify, request, Response
+from flask_swagger_ui import get_swaggerui_blueprint
+from utils import get_connection, create_fetch_cmd, create_cmd_workspace,\
+    create_cmd_up_pinfile, check_workspace_empty
+
 
 app = Flask(__name__)
 
+APP_DIR = os.path.dirname(os.path.realpath(__file__))
 # Reading directory path from config.yml file
+with open(APP_DIR + '/config.yml', 'r') as f:
+    config = yaml.load(f)
 
-with open('config.yml', 'r') as f:
-    doc = yaml.load(f)
-
-with open('swagger.json', 'r') as f:
-    jsonData = json.load(f)
-
-WORKING_DIR = doc['working_path']
-LOGGER_FILE = doc['logger_file_name']
-DB_PATH = doc['db_path']
-INVENTORY_PATH = doc['inventory_path']
-LATEST_PATH = doc['linchpin_latest_file_path']
-PINFILE_JSON_PATH = doc['pinfile_json_path']
+WORKSPACE_DIR = config.get('workspace_path', '/tmp/')
+LOGGER_FILE = config.get('logger_file_name', 'restylinchpin.log')
+DB_PATH = config.get('db_path', 'db.json')
+INVENTORY_PATH = config.get('inventory_path', '/dummy/inventories/*')
+LATEST_PATH = config.get('linchpin_latest_file_path',
+                         '/dummy/resources/linchpin.latest')
+PINFILE_JSON_PATH = config.get('pinfile_json_path', '/dummy/PinFile.json')
```

Comment:
Is there any specific reason that we are using this method to read data from yaml files asking because it needs hardcoding and we have specified these paths in the config.yml file as well so it is kind of repetitive.

Reply:
@mansikulkarni96 It is repetitive to be safe by default. 
Let us assume the config file gets corrupted or deleted. In that case, the server might not start and also it might affect already running server servers. 
Thus providing defaults might help us.

Note: 
I have defaulted config variable to {} in case of file not found